pkg/generator/grafana/grafana.go

```diff
@@ -100,24 +90,66 @@ func (w *Grafana) Cleanup(ctx context.Context, jsonSpec *apiextensions.JSON, pre
 }
 
 func newClient(ctx context.Context, gen *genv1alpha1.Grafana, kclient client.Client, ns string) (*grafanaclient.GrafanaHTTPAPI, error) {
-	secret, err := resolvers.SecretKeyRef(ctx, kclient, resolvers.EmptyStoreKind, ns, &esmeta.SecretKeySelector{
-		Namespace: &ns,
-		Name:      gen.Spec.Auth.Token.Name,
-		Key:       gen.Spec.Auth.Token.Key,
-	})
-	if err != nil {
-		return nil, err
-	}
 	url := strings.TrimPrefix(gen.Spec.URL, "https://")
+
+	scheme := "https"
+	if strings.HasPrefix(url, "http://") {
+		scheme = "http"
+	}
+
+	// Trim the scheme from the URL
+	if strings.Contains(url, "://") {
+		_, url, _ = strings.Cut(url, "://")
+	}
+
 	cfg := &grafanaclient.TransportConfig{
 		Host:     url,
 		BasePath: "/api",
-		Schemes:  []string{"https"},
-		APIKey:   secret,
+		Schemes:  []string{scheme},
+	}
```

Comment:
Looking at this, i think it would make sense to use `url.Parse()` instead and pass scheme/host/path into `TransportConfig`.

Reply:
Good call. I've switched to calling parsing the URL, and I've updated the client BasePath to use a path prefix from the URL, if any is provided (e.g. `http://services.my-domain.com/grafana` will set `BasePath` to `/grafana/api`). This is somewhat common when services are hosted under a single domain name, with a reverse proxy in front.